### ISSUE TITLE
Use specific "llama.cpp" version

### ISSUE THREAD

**Issue by ILYA-2606** [ShenghaiWang/SwiftLlama#16]
```swift
.package(url: "https://github.com/ggerganov/llama.cpp.git", branch: "master")
```
It is better to use a special version of the package, because master is constantly updated and different revisions may work unstable or not work at all even

**Comment by ShenghaiWang** [comment]
The llama.cpp package contains `unsafeFlags`, which prevents from using semantic versions.

Please refer to https://forums.swift.org/t/override-for-unsafeflags-in-swift-package-manager/45273/5 for detailed explanation and some workaround.

**Comment by jaekong** [comment]
Currently it's not working because llama.cpp got rid of Package.swift.

**Comment by meaning-matters** [comment]
> Currently it's not working because llama.cpp got rid of Package.swift.

One can build XCFramework with `llama.cpp`'s `build-xcframework.sh` and use that instead of `llama.cpp` in `Package.swift`, or get it from a release's assets (e.g. https://github.com/ggml-org/llama.cpp/releases/tag/b5142).

**Comment by jaekong** [comment]
Yeah it's possible to work around it, but it's not ideal - especially for Linux + macOS multiplatform codebase.

**Comment by ShenghaiWang** [comment]
Can use v0.4.0 to solve this issue where we use precompiled framework
